VultisigApp/VultisigApp/View Models/KeysignViewModel.swift

```diff
@@ -88,21 +88,23 @@ class KeysignViewModel: ObservableObject {
 
     func startKeysign() async {
         defer {
-            self.messagePuller?.stop()
+            messagePuller?.stop()
         }
-        for msg in self.messsageToSign {
+        for msg in messsageToSign {
             do {
+                try await verifyAllowance()
```

Comment:
Could you please do this check before keysign? Like in confirm screen? 

Because if we check here , and the approval is not committed ,then user might need to refill the form again

Reply:
I we'll check this before keysing that check will always failed because transaction just sent